kernel/src/main/java/org/kframework/kompile/KompileFrontEnd.java

```diff
@@ -68,6 +68,7 @@ public int run() {
         Kompile kompile = new Kompile(options, files.get(), kem, sw, !options.profileRules);
         Backend backend = koreBackend.get();
         CompiledDefinition def = kompile.run(options.outerParsing.mainDefinitionFile(files.get()), options.mainModule(files.get()), options.syntaxModule(files.get()), backend.steps(), backend.excludedModuleTags());
+        files.get().saveToKompiled("mainModule.txt", def.executionModule().name());
         sw.printIntermediate("Kompile to kore");
```

Comment:
This information isn't already saved somewhere else?

Reply:
It is not saved anywhere in any place that is convenient for a shell script to parse. Mostly the java code gets this information from the deserialized K definition, which we are not making use of in the new script.